test/IntelFPGAMemoryAttributes.ll

```diff
@@ -11,6 +11,8 @@
 ; CHECK-SPIRV: Decorate {{[0-9]+}} NumbanksINTEL 4
 ; CHECK-SPIRV: Decorate {{[0-9]+}} BankwidthINTEL 8
 ; CHECK-SPIRV: Decorate {{[0-9]+}} MaxconcurrencyINTEL 4
+; CHECK-SPIRV: Decorate {{[0-9]+}} SinglepumpINTEL 1
```

Comment:
Strictly speaking, this test violates the [spec](https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/INTEL/SPV_INTEL_fpga_memory_attributes.asciidoc)

As I can see, `RegisterINTEL`, `SinglepumpINTEL` and `DoublepumpINTEL` have **no** extra operands and you inserting one [integer literal](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_terms) here.

I guess it happened because you re-used existing infrastructure for these attributes introduced in d51037eb

This may cause problems if someone will use another SPIR-V consumer (which corresponds to the spec) instead of this translator.
Probably it is not critical, but I would like to see it fixed either in the spec or here in translator

Reply:
Good spot, thank you! This will be discussed with the spec authors, and the necessary adjustments will be made within May.